src/files/examples/fingerprint_reader_functions.py

```diff
@@ -0,0 +1,260 @@
+
+	except Exception as e:
+		print('Operation failed!')
+		print('Exception message: ' + str(e))
+		exit(1)
+
+def search():
+	"""Searches for given fingerprint"""
+	## Tries to search the finger and calculate hash
+	try:
+		print('Waiting for finger...')
+
+		## Wait that finger is read
+		while ( f.readImage() == False ):
+			pass
+
+		## Converts read image to characteristics and stores it in charbuffer 1
+		logging.debug("Image read converting it")
+		f.convertImage(0x01)
+
+		## Searchs template
+		result = f.searchTemplate()
+
+		positionNumber = result[0]
+		accuracyScore = result[1]
+
+		if ( positionNumber == -1 ):
+			print('No match found!')
+			exit(0)
+		else:
+			print('Found template at position #' + str(positionNumber))
+			print('The accuracy score is: ' + str(accuracyScore))
+
+		## OPTIONAL stuff
+		##
+
+		## Loads the found template to charbuffer 1
+		f.loadTemplate(positionNumber, 0x01)
+
+		## Downloads the characteristics of template loaded in charbuffer 1
+		characteristics = str(f.downloadCharacteristics(0x01)).encode('utf-8')
+		logging.debug("Characteristics raw:%s" % (characteristics))
+
+		## Hashes characteristics of template
+		print('SHA-2 hash of template: ' + hashlib.sha256(characteristics).hexdigest())
+
+	except Exception as e:
+		print('Operation failed!')
+		print('Exception message: ' + str(e))
+		exit(1)
+
+
+def init_arg_parse():
+	"""App help"""
+	#https://docs.python.org/2/library/optparse.html
+	#https://docs.python.org/2/library/argparse.html#printing-help
+	#https://docs.python.org/2/howto/argparse.html#id1
+	global parser
+	parser = argparse.ArgumentParser(description='Do the given action with fingerprintReader')
+	
+	parser.add_argument("-v", "--verbose", help="Display system parameters of sensor", action="store_true")
+	parser.add_argument("-e", "--enroll", help="Create a new fingerprint model and store it into database", action="store_true")
+	parser.add_argument("-t", "--templates", help="Display current templates count", action="store_true")
+	parser.add_argument("-d", "--delete", help="Deletes a template after getting fingerprints", action="store_true")
+	parser.add_argument("-D", "--delete_all_templates", help="Deletes all templates of database", action="store_true")
+	parser.add_argument("-i", "--download_image", help="Downloads image and put it into tempDir", action="store_true")
+	parser.add_argument("-s", "--search", help="Searches for given fingerprint into database and display if found", action="store_true")
+	args = parser.parse_args()
+	if args.verbose:
+		display_system_parameters()
+	elif args.enroll:
+		enroll()
+	elif args.delete:
+		delete()
+	elif args.delete_all_templates:
+		delete_all()
+	elif args.download_image:
+		download_image()
+	elif args.search:
+		search()
+	elif args.templates:
+		logging.info("Currently used templates %s/%s" % (f.getTemplateCount(), f.getStorageCapacity()))	
+	else:
+		parser.print_help()
+
+
+def enroll():
```

Comment:
why this method is below of `init_arg_parse()`?

Reply:
you're perfectly right